File: backend/TemperatureAndHumidityLogger.Backend/TemperatureAndHumidityLogger.Core/Entities/Devices/Device.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using TemperatureAndHumidityLogger.Core.Entities.Logs;
using TemperatureAndHumidityLogger.Core.Entities.Users;
using TemperatureAndHumidityLogger.Core.Interfaces.Common;

namespace TemperatureAndHumidityLogger.Core.Entities.Devices
{
    public class Device : BaseEntity, ISoftDelete
    {
        public Guid SerialNumber { get; set; }
        public string Caption { get; set; }
        public DateTime? DeletedAt { get; set; }

        [ForeignKey("Users")]
        public Guid? UserId { get; set; }
        public User User { get; set; }
        public List<Log> Logs { get; set; }

        public float? MinTemperature { get; set; }
        public float? MaxTemperature { get; set;}
        public float? MinHumidity { get; set; }
        public float? MaxHumidity { get;set; }
        public DateTime? LastMessageDate { get; set; }
    }
}

File: ui/temperature-ui/src/entities/device.ts
export interface Device {
  id?: string;
  serialNumber: string;
  caption: string;
  deletedAt?: Date;
  userId?: string;
  minTemperature?: number;
  maxTemperature?: number;
  minHumidity?: number;
  maxHumidity?: number;
  lastMessageDate?: Date;
}

export interface AssignDeviceRequest {
  serialNumber?: string;
}

export interface UpdateDeviceRequest {
  serialNumber: string; 
  caption: string;
  minTemperature?: number;
  maxTemperature?: number;
  minHumidity?: number;
  maxHumidity?: number;
}

